src/store/modules/util/actions.ts

```diff
@@ -26,6 +27,118 @@ const actions: ActionTree<UtilState, RootState> = {
     } catch (err) {
       console.error("error", err);
     }
+  },
+  async getFacilities({ commit, state, dispatch }, payload) {
+    let resp;
+
+    try{
+      const filters = state.currentLocationFilterSelected;
+      if (filters.productStoreId !== 'All') payload.inputFields.productStoreId = filters.productStoreId;
+      if (filters.facilityTypeId !== 'All') payload.inputFields.facilityTypeId = filters.facilityTypeId;
+
+      resp = await UtilService.getFacilities(payload);
+
+      if(resp.status === 200 && resp.data.docs?.length && resp.data.docs?.length > 0 && !hasError(resp)) {
+        let facilities = resp.data.docs;
+        const total = resp.data.count;
+
+        let facilityIds: any = new Set();
+        facilities.forEach((facility: any) => {
+          if (facility.facilityId) facilityIds.add(facility.facilityId);
+        });
+        facilityIds = [...facilityIds]
+
+        await dispatch('getFacilityIdentifications', facilityIds).then((res) => {
+          facilities.map((facility: any) => {
+            const facIdent = res.find((fac: any) => fac.facilityId === facility.facilityId);
+
+            if(facIdent?.facilityIdenTypeId === "SHOPIFY_FAC_ID") {
+              facility.shopifyId = facIdent?.idValue;
+            } else if(facIdent?.facilityIdenTypeId === "NETSUITE_FAC_ID") {
+              facility.netsuiteId = facIdent?.idValue;
+            }
+          })
+        });
+        await dispatch('getFacilityAddress', facilityIds).then((res) => {
+          facilities.map((facility: any) => {
+            const facAddress = res.find((fac: any) => fac.facilityId === facility.facilityId);
+
+            facility.address1 = facAddress?.address1;
+            facility.address2 = facAddress?.address2;
+            facility.city = facAddress?.city;
+            facility.postalCode = facAddress?.postalCode;
+            facility.stateGeoCode = facAddress?.stateGeoCode;
+          })
+        });
+        if (payload.viewIndex && payload.viewIndex > 0) facilities = state.facilityLocations.list.concat(facilities);
+        commit(types.UTIL_FACILITY_LOCATIONS_UPDATED, { facilities, total });
+      } else {
+        showToast(translate("Something went wrong"));
+        commit(types.UTIL_FACILITY_LOCATIONS_UPDATED, { facilities: [], total: 0});
+      }
+    } catch(error) {
+      showToast(translate("Something went wrong"));
+      console.error(error);
+    }
+    return resp;
+  },
+  async getFacilityIdentifications({ commit }, payload) {
+    let resp;
+
+    try{
+      const params = {
+        "inputFields": {
+          "facilityId": payload,
+          "facilityId_op": "in",
+          "facilityIdenTypeId": ["SHOPIFY_FAC_ID", "NETSUITE_FAC_ID"],
+          "facilityIdenTypeId_op": "in"
+        },
+        "viewSize": payload.length,
```

Comment:
I think this should be twice the length, we are getting 2 identifications

Reply:
Okay Sir, improved the code to increase the length of viewSize by multiplying by 2.